index.bs

```diff
@@ -2723,6 +2770,218 @@ The [=remote end steps=] with <var ignore>session</var> and |command parameters|
 
 1. Return [=success=] with data null.
 
+</div>
+#### The browsingContext.locateNodes Command ####  {#command-browsingContext-locateNodes}
+
+The <dfn export for=commands>browsingContext.locateNodes</dfn> command returns a
+list of all nodes matching the specified locators.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+      <pre class="cddl remote-cddl">
+      browsingContext.LocateNodes = (
+        method: "browsingContext.locateNodes",
+        params: browsingContext.LocateNodesParameters
+      )
+
+      browsingContext.LocateNodesParameters = {
+         context: browsingContext.BrowsingContext,
+         locator: browsingContext.Locator,
+         startNodes: [ * script.SharedReference ],
+         ? maxNodes: js-uint,
+         ? ownership: script.ResultOwnership,
+         ? serializationOptions: script.SerializationOptions
+      }
+      </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+        browsingContext.LocateNodesResult = {
+            context: browsingContext.BrowsingContext,
+            nodes: [ * script.NodeRemoteValue ]
+        }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="locate nodes using CSS">
+To <dfn>locate nodes using CSS</dfn> with given |context target|, |context nodes|,
+|selector| and <var ignore>session</var>:
+
+1. Let |returned nodes| be an empty array.
+
+1. For each |context node| of |context nodes|:
+
+  1. Let |parse result| be the result of [=parse a selector=] given |selector|.
+
+  1. If |parse result| is failure, return [=error=] with [=error code=] [=invalid selector=].
```

Comment:
Why do we have to do parse the selector for each and every context node? It stays the same and as such we can move this out to just before the loop.

The same applies to the other methods as well.

Reply:
Good point. I don't think the other algorithms suffer from the same issue. For XPath, we have to call `evaluate` using the selector on each of the context nodes, so there is no "parse the XPath to make sure it's valid" step before calling `evaluate`. For find by text, the regexp is parsed before entering the loop for the context nodes.